api/src/auth/auth.ts

```diff
@@ -11,13 +11,13 @@ export abstract class AuthDriver {
 	}
 
 	/**
-	 * Get user id by unique identifier
+	 * Get user id for a given provider payload
 	 *
-	 * @param identifier Unique user identifier
+	 * @param payload Any data that the user might've provided
 	 * @throws InvalidCredentialsException
 	 * @return User id of the identifier
 	 */
-	abstract userID(identifier: string): Promise<string>;
+	abstract getUserID(identifier: string | number, payload: Record<string, any>): Promise<string>;
```

Comment:
Why not kill `identifier` and only have `payload`?

Reply:
Figured `identifier` is a known standard in that part of the auth flow, while the payload could be any user provided thing, that makes identifier slightly more reliable

Reply:
Hm... I dunno. I kinda like the idea of going full payload. Currently I use the `code` from the oauth response as the identifier, but it's kinda a misnomer

Reply:
Works for me!